### PR TITLE
Init inventory page

### PR DESCRIPTION
**Pull request by Monnte**
**OAMG-9571**

MVP for inventory page highly inspired from [tasks-ui](https://github.com/RedHatInsights/tasks-frontend/tree/8772a8565f232aa3ea2767b4d0c16eb3ff02226d/src/SmartComponents/SystemTable). Implemented solution supports only basic functionality of InventoryTable since it is an MVP.

- [x] PR has a short description
- [ ] Tests for the changes have been added

### CODE REVIEW

File: src/AppEntry.tsx [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-  <Provider store={init(...(process.env.NODE_ENV !== 'production' ? [logger] : [])).getStore()}>
-    <Router basename={getBaseName(window.location.pathname)}>
-      <App />
-    </Router>
-  </Provider>
-);
+const AppEntry = () => {
+  const registry = init(...(process.env.NODE_ENV !== 'production' ? [logger] : []));
+  return (
+    <RegistryContext.Provider value={{ getRegistry: () => registry }}>
```
Comment (Wecros) [inline src/AppEntry.tsx:12]: What is the purpose of the added RegistryContext?

**Comment by Wecros** [comment]
LGTM 👍 I tested the inventory and works expectedly. One thing I would add is that the text of the system hostnames looks too small to me, but that can be adjusted later.

**Review by Wecros — APPROVED** [review]

File: src/AppEntry.tsx (hunk 1 above)
Comment (Monnte) [reply src/AppEntry.tsx:12]: To access registry from InventoryPage component and then do some magic copied from tasks repo to make component work :smile:

**Comment by Monnte** [comment]
Tests are not working due some error that will be fixed in some another PR.

Context:
```
    import * as _ from 'lodash-es';
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module

```

**Comment by andywaltlova** [comment]
Alright the problems are permissions, talked with @Monnte ,we will try using  https://github.com/RedHatInsights/frontend-components/blob/master/packages/components/src/RBACProvider/RBACProvider.tsx with combinations of https://github.com/RedHatInsights/frontend-starter-app/blob/master/src/Routes/NoPermissionsPage/NoPermissionsPage.tsx , probably copying what is in https://github.com/RedHatInsights/vulnerability-ui/blob/master/src/Components/SmartComponents/SystemsPage/SystemsPage.js#L57 ... we will see how that will go


MVP won't definitely has it's own backend, we need at least simple UI message that permisisons are misisng instead of just crashing - I am not saying that we need it right now, can be separate if deemed too complicated, but I doubt they will let us deploy with this

File: src/api.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+export const TASKS_API_ROOT = '/api/tasks/v1';
```
Comment (andywaltlova) [inline src/api.ts:3]: Also talked with @Monnte about why tasks api is used, I know we do not have our backend but I though we will just do 1:1 copy of inventory table.. meaning use same api as they do (https://prod.foo.redhat.com:1337/api/inventory/v1/hosts?per_page=50&page=1 ...) and if we are in need of any additional filtering we could just decide to 'not render' systems that are not usable for ugprades page (e.g. 9.Y systems)

@Wecros WDYT? so far we agreed with @Monnte that he will change it for 1:1 copy with inventory api

**Review by andywaltlova — CHANGES_REQUESTED** [review]

File: src/api.ts (hunk 2 above)
Comment (Monnte) [reply src/api.ts]: Already changed to inventory api :+1:

File: src/Components/InventoryPage/InventoryPage.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+          <Spinner />
+        </Bullseye>
+      }
+      onLoad={defaultOnLoad(getRegistry)}
+      getEntities={getEntities}
+      tableProps={{
+        isStickyHeader: true,
+        isStriped: true,
+        canSelectAll: false,
+        onSelect: items.length ? selectIds : null,
```
Comment (andywaltlova) [inline src/Components/InventoryPage/InventoryPage.tsx:140]: I noticed there is some difference between our table and inventory, if you click checkbox to select all, it does many... and I mean many requests to select all 4k systems (when I am logged on my profile) but inventory table selects only the first page.

image

Was this intentional?

**Review by andywaltlova — COMMENTED** [review]
Apart from that one comment, it's looking solid. Tried to debug the tests, but I am not seeing anything, I would be then in favor of removing the tests at all and create separate task for fixing the test suit that can be done asynchronously with implementations

File: src/Components/InventoryPage/InventoryPage.tsx (hunk 3 above)
Comment (Monnte) [reply src/Components/InventoryPage/InventoryPage.tsx:140]: Nope, I will change it to page select :+1:

File: src/api.ts (hunk 2 above)
Comment (Wecros) [reply src/api.ts]: agree with the change

**Review by Wecros — APPROVED** [review]
lgtm after the changes, nice job 👍

**Review by andywaltlova — COMMENTED** [review]
Page fix looks good, can be merged after the test is disabled and task for resolving it is created

**Review by andywaltlova — APPROVED** [review]

**Pull request merged**